src/internal/pjs/api_server.go

```diff
@@ -111,3 +112,92 @@ func (a *apiServer) InspectJob(ctx context.Context, req *pjs.InspectJobRequest)
 		},
 	}, nil
 }
+
+func (a *apiServer) ProcessQueue(srv pjs.API_ProcessQueueServer) (retErr error) {
+	ctx := srv.Context()
+	req, err := srv.Recv()
+	if err != nil {
+		return errors.EnsureStack(err)
+	}
+	if req.Queue == nil {
+		return status.Errorf(codes.InvalidArgument, "first message must pick Queue")
+	}
+
+	for {
+		var jobID pjsdb.JobID
+		if err := dbutil.WithTx(ctx, a.env.DB, func(ctx context.Context, sqlTx *pachsql.Tx) error {
```

Comment:
note to self, I think we decided this was okay for now since we didn't want to pass in the transaction server as a dependency.